### PR TITLE
Backend/900/dev group

### PR DESCRIPTION
**Pull request by riatzukiza**
### Summary of Changes

Resolves issue #900

Added `ENABLE_DEVELOPER_GROUP` to settings. By default, this value is true, in production it is false.

#### Test the Developer user group is added when in development or staging mode

1. Start server in development mode (`cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d`)
2. As super user, open the admin panel, select groups and confirm Developer is among them
3. Navigate back to the dashboard, and select users
4. Select your self from among the users, add them to the developer group and remove them from any other group
5. Save changes, if you were successful, you should see and use the admin dashboard just as you would before.
6. Repeat steps on an app started in staging mode

#### Test the Developer user group is not present when in production Mode

1. Start the server in prodiuction mode (Add command or be more specific)
2. As a super user, open the admin panel, select groups, and confirm Developer is not amoung them

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [ ] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [ ] Check against the criteria:

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [ ] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

+ [ ] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [ ] If this PR introduces backend code, is that code documented both inline and overall?
+ [ ] If this PR introduces frontend code, is that code documented both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any security issues?

### CODE REVIEW

File: tdrs-backend/tdpservice/users/migrations/0021_add_dev_group.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
+from django.db import migrations
+from django.contrib.auth.models import Group
+from django.conf import settings
+
+from tdpservice.users.permissions import delete_permissions_q
```
Comment (jtwillis92) [inline tdrs-backend/tdpservice/users/migrations/0021_add_dev_group.py:6]:
```suggestion
```
This import is unused.

**Review by jtwillis92 — APPROVED** [review]

**Review by abottoms-coder — APPROVED** [review]

File: tdrs-backend/tdpservice/settings/common.py [hunk 2]
```diff
@@ -359,3 +359,5 @@ class Common(Configuration):
         'OIDC_OP_TOKEN_ENDPOINT',
         'https://idp.int.identitysandbox.gov/api/openid_connect/token'
     )
+
+    ENABLE_DEVELOPER_GROUP = True
```
Comment (ADPennington) [inline tdrs-backend/tdpservice/settings/common.py:363]:
```suggestion
    ENABLE_DEVELOPER_GROUP = False
```

File: tdrs-backend/tdpservice/settings/common.py [hunk 3]
```diff
@@ -359,3 +359,5 @@ class Common(Configuration):
         'OIDC_OP_TOKEN_ENDPOINT',
         'https://idp.int.identitysandbox.gov/api/openid_connect/token'
     )
+
+    ENABLE_DEVELOPER_GROUP = False
```
Comment (ADPennington) [inline tdrs-backend/tdpservice/settings/common.py:363]:
```suggestion
    ENABLE_DEVELOPER_GROUP = True
```

**Review by ADPennington — APPROVED** [review]
thanks @riatzukiza 🚀 @abottoms-coder ready to merge

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
   + [x] Developer group exists in dev and staging - this gives dev super user without needing PIV/CAC
   + [x] A user in the developer group in dev or staging does not need a PIV/CAC card in order to high level privilege
   + [x]  Developer group does NOT exist in production

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

n/a

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [x] If this PR introduces backend code, is that code documented both inline and overall?


### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?

no new issues

**Pull request merged**